### PR TITLE
libretro.pcsx2: init at unstable-2021-11-27

### PR DESCRIPTION
**Pull request by thiagokokada**
###### Motivation for this change

I was having difficult to package this core on #146958, however with the help of @jnetod I finally got it right.

Tested and working.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
Help wanted, it is failing because it can't find the library: http://ix.io/3GhS

**Comment by thiagokokada** [comment]
I tried to list all files from `/build` directory: http://ix.io/3Gi2

It is not building the `libretro_pcsx2.so` file for some reason. I think it is missing some `cmakeFlags` to enable building a library. However I don't have a clue what this flag should be.

I am basing the current `cmakeFlags` from those:
- https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=libretro-pcsx2-git
- https://docs.libretro.com/development/cores/core-specific/pcsx2/

Also, I tried to use `cmake --build` instead `Makefile` too, however it still didn't work (same issue). And `cmake --build . --target pcsx2_libretro --config Release` doesn't work because `pcsx2_libretro` is missing.

**Comment by thiagokokada** [comment]
CC @aanderse

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 147622` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libretro.pcsx2</li>
    <li>retroarchFull</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Tested with Tekken Tag Tournament:

Tekken Tag Tournament (USA) (v2 00)-211128-132952

File: pkgs/misc/emulators/retroarch/cores.nix [hunk 1]
```diff
@@ -669,6 +673,36 @@ in
...
+      libGLU
+      libpcap
+      libpng
+      libxml2
+      xz
+      xxd
+    ];
+    makefile = "Makefile";
+    cmakeFlags = [
+      "-DCMAKE_BUILD_TYPE=Release"
```
Comment (aanderse) [inline pkgs/misc/emulators/retroarch/cores.nix:696]: This should be the default already.

**Review by aanderse — CHANGES_REQUESTED** [review]
Awesome!

**Comment by thiagokokada** [comment]
@aanderse Done.

**Review by aanderse — APPROVED** [review]
LGTM 👍

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 147622` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>libretro.dolphin</li>
    <li>libretro.pcsx2</li>
    <li>libretro.swanstation</li>
    <li>retroarchFull</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 147622` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>libretro.beetle-saturn</li>
    <li>libretro.dolphin</li>
    <li>libretro.flycast</li>
    <li>libretro.pcsx2</li>
    <li>libretro.swanstation</li>
    <li>retroarchFull</li>
  </ul>
</details>

**Pull request merged**